src/components/SchemaResourcePage/SchemaResourceEdit.tsx

```diff
@@ -66,10 +72,24 @@ export function SchemaResourceEdit({
   }
   watch("spec");
 
+  const hasNamespaceField = useMemo(() => {
```

Comment:
Shouldn't this be part of the resource definition ?

Reply:
Yeah, that is the right place for this logic. I'll refactor this accordingly.